### ISSUE TITLE
Support -Crelocation-model

### ISSUE THREAD

**Issue by bjorn3** [rust-lang/rustc_codegen_gcc#205]
This is necessary for compiling rust-for-linux with cg_gcc.

**Comment by antoyo** [comment]
It seems the LLVM codegen uses `LLVMRustSetDSOLocal`: I'm not exactly sure what is the GCC equivalent.

Adding the GCC flags `-mcmodel=kernel -fno-pie` when relocation_model is static allow compiling rust-for-linux with cg_gcc. But I'm not sure whether this would be correct.

**Comment by bjorn3** [comment]
> Adding the GCC flags -mcmodel=kernel -fno-pie when relocation_model is static allow compiling rust-for-linux with cg_gcc. But I'm not sure whether this would be correct.

Yeah, that worked for me too. There is a bug somewhere in how kernel module parameters are codegened causing the respective sample to crash during early boot though.

The code model option can be directly translated to `-mcmodel=<code model>` I think. The relocation model will probably need a match and then pass the appropriate `-f` argument corresponding to LLVM's relocation model.

**Comment by antoyo** [comment]
My other change to allow running the kernel modules is to remove all the SIMD flags.

**Comment by bjorn3** [comment]
I added the `-mno-...` flags that are passed to GCC when by linux when compiling C code. That worked too with the exception of disabling sse. Even with the soft float abi it complains.

**Comment by antoyo** [comment]
Implemented in #578.
